content/blog/choose-tidymodels-adventure/index.Rmd

```diff
@@ -0,0 +1,81 @@
+
+## Starting with the basics: **parsnip**
+
+A fully featured ecosystem for modeling and machine learning requires interfaces to, well, models, and the tidymodels package that provides those functions and interfaces is [parsnip](https://parsnip.tidymodels.org/). If your modeling adventure involves small data and straightforward data preprocessing (like that provided by [R's model formula](https://www.tmwr.org/base-r.html#formula)), you may be well-served by focusing on parsnip. To learn more about how to fit and evaluate parsnip models, check out [this article at tidymodel.org's **Get Started** section](https://www.tidymodels.org/start/models/) and [this blog post (complete with screencast) by me](https://juliasilge.com/blog/student-debt/).
+
+![Student loan debt by race across time](https://juliasilge.com/blog/student-debt/index_files/figure-html/unnamed-chunk-3-1.png)
+
+We don't believe most people using tidymodels fall into this first category, but we think that the tools we've built for these kinds of straightforward analyses are well-designed and will set you up for statistical and practical success. Some people will not reach for tidymodels as a whole ecosystem or even parsnip, choosing instead to use underlying model functions, like `lm()` instead of `linear_reg() %>% set_engine("lm")` (or at another extreme of data size and model complexity, the flexibility of [keras](https://keras.rstudio.com/) for deep learning architectures). Why might you choose to use one or more tidymodels packages anyway, if parsnip or the whole ecosystem is not a good fit for you?
+
+- You may want to use our tidy data resampling infrastructure together with non-tidymodels modeling functions, as shown [here](https://www.tidymodels.org/learn/statistics/bootstrap/) and [here](https://juliasilge.com/blog/ceo-departures/).
+
+- You may want to use [recipes](https://recipes.tidymodels.org/) for feature engineering with non-tidymodels modeling functions, as shown [here](https://smltar.com/dldnn.html).
+
+## Holistic model **workflows**
+
+When you are setting off on a modeling adventure, it might be worth asking what we even mean by the word "model"; it is a word that gets overloaded really quickly! In the tidymodels ecosystem, we carefully incorporate *both* feature engineering (also called data preprocessing) that must be learned from training data *and* a model fit into a modeling workflow that is estimated together. For example, if you trained a least squares regression model with features learned from principal component analysis, the PCA preprocessing step should be considered part of the model workflow:
+
+![PCA preprocessing is part of the modeling process](https://www.tmwr.org/premade/proper-workflow.svg)
```

Comment:
I think each of the figures could use either a caption or a sentence in the post that directly links to the figure (e.g. "as shown in the figure below..." so it's clear how they relate to the rest of the post).